build-system/release-workflows/upload-release.js

```diff
@@ -100,6 +101,54 @@ function logProgress_(totalFiles, uploadedFiles) {
   }
 }
 
+/**
+ * Compresses a single file with Brotli and writes it to ${path}.br.
+ * @param {string} path
+ * @param {number} sizeHint
+ * @return {Promise<void>}
+ */
+async function brotliCompressFile_(path, sizeHint) {
+  const fileBuffer = fs.readFileSync(path);
+  const compressedBuffer = zlib.brotliCompressSync(fileBuffer, {
+    params: {
+      [zlib.constants.BROTLI_PARAM_QUALITY]: zlib.constants.BROTLI_MAX_QUALITY,
+      [zlib.constants.BROTLI_PARAM_SIZE_HINT]: sizeHint,
+    },
+  });
+  return fs.writeFileSync(`${path}.br`, compressedBuffer);
+}
+
+/**
+ * Compresses all files with Brotli.
+ * @return {Promise<void>}
+ */
+async function brotliCompressAll_() {
+  let totalFiles = 0;
+  for await (const {stats} of klaw(DEST_DIR)) {
+    if (stats.isFile()) {
+      totalFiles++;
+    }
+  }
+
+  log('Brotli compression of', cyan(totalFiles), 'files:');
+  const compressPromises = [];
+  let compressedFiles = 0;
+  for await (const {path, stats} of klaw(DEST_DIR)) {
+    if (!stats.isFile()) {
+      continue;
+    }
+
+    compressPromises.push(
+      brotliCompressFile_(path, stats.size).then(() => {
+        logProgress_(totalFiles, ++compressedFiles);
+      })
+    );
+  }
+
+  await Promise.all(compressPromises);
```

Comment:
Error handling? Should the CI job fail if compression fails?

Reply:
I think it should, this is going to become an important part of the CDN soon and it's not a piece of code that's likely to fail often, so if it does it probably means something very bad happened